scipy/spatial/transform/tests/test_rigid_transform.py

```diff
@@ -1011,6 +1011,65 @@ def test_input_validation(xp):
         RigidTransform.from_rotation(xp.eye(3))
 
 
+@make_xp_test_case(RigidTransform.mean)
+@pytest.mark.parametrize("ndim", range(1, 4))
+def test_mean(xp, ndim: int):
+    atol = 1e-12
+    rng = np.random.default_rng(123)
+
+    dtype = xpx.default_dtype(xp)
+    t = xp.asarray(rng.normal(size=(ndim,) * (ndim - 1) + (3,)), dtype=dtype)
+    q = xp.asarray(rng.normal(size=(ndim,) * (ndim - 1) + (4,)), dtype=dtype)
+    r = rotation_to_xp(Rotation.from_quat(q), xp=xp)
```

Comment:
Should not be necessary since Rotation was already created from an xp array of correct dtype.